### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by AgazadeAV**

### CODE REVIEW

File: hotel/src/main/java/com/andersenhotels/model/Hotel.java [hunk 1]
```diff
@@ -1,59 +1,84 @@
...
-            throw new IllegalStateException(message);
+    public Map<Integer, Reservation> getReservationMap() {
+        Map<Integer, Reservation> map = new HashMap<>();
+        for (Reservation reservation : reservations) {
+            map.put(reservation.getApartment().getId(), reservation);
         }
-        reservations.remove(id);
-        LOGGER.info("Reservation removed successfully for Apartment ID = {}", id);
+        return map;
     }
```
Comment (Alexmansar) [inline hotel/src/main/java/com/andersenhotels/model/Hotel.java]: a lot of logic in entitty class. use service for it

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@AgazadeAV Why all tests were removed?
you have several entities
create 1 interface with the base CRUD opearation, and create impl for each service and just use it)

**Comment by Alexmansar** [comment]
> Why all tests were removed?

@AgazadeAV Hi)) so, why all tests were removed?

**Comment by AgazadeAV** [comment]
@Alexmansar, I'm rewriting them again, because all logic has been changed and I decided that it will be easier to write them again))
right now I'm writing tests))

File: hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Mock
+    private EntityTransaction transaction;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java]: use annotation style please

File: hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        when(entityManagerFactory.createEntityManager()).thenReturn(entityManager);
+        when(entityManager.getTransaction()).thenReturn(transaction);
+        doNothing().when(transaction).rollback();
+        service = new TestCrudService(entityManagerFactory, TestEntity.class);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Mockito.reset(entityManager, transaction, entityManagerFactory);
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java]: static import

File: hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+import jakarta.persistence.*;
+import lombok.Getter;
+import org.junit.jupiter.api.*;
+import org.mockito.*;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/AbstractCrudServiceTest.java:15]: I mean you use it here, so in other places withou Mockito.

File: hotel/src/test/java/com/andersenhotels/model/service/GuestServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+    private GuestService guestService;
+
+    @Mock
+    private Guest mockGuest;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        guestService = spy(new GuestService());
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/GuestServiceTest.java]: same about annotations

File: hotel/src/test/java/com/andersenhotels/model/service/ReservationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        apartmentService = mock(ApartmentService.class);
+        reservationService = spy(new ReservationService() {
+            @Override
+            void updateApartmentStatus(Apartment apartment, ApartmentStatus status) {
+                apartment.setStatus(status);
+                apartmentService.update(apartment);
+            }
+        });
+        guest = mock(Guest.class);
+        apartment = mock(Apartment.class);
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/ReservationServiceTest.java]: and here

File: hotel/src/test/java/com/andersenhotels/model/service/ReservationServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+//        when(reservation.getId()).thenReturn(1);
+//        doReturn(true).when(reservationService).existsById(1);
+//        when(reservation.getApartment()).thenReturn(apartment);
+//
+//        reservationService.cancelReservation(reservation);
+//
+//        verify(reservationService, times(1)).delete(1);
+//        verify(reservationService, times(1)).updateApartmentStatus(apartment, ApartmentStatus.AVAILABLE);
+//        verify(apartmentService, times(1)).update(apartment);
+//    }
```
Comment (Alexmansar) [inline hotel/src/test/java/com/andersenhotels/model/service/ReservationServiceTest.java]: remove or use

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@AgazadeAV Hi)) please, use annotation style when work with Mockito

**Review by Alexmansar — APPROVED** [review]

**Pull request merged**